packages/headless/src/api/generated-answer/generated-answer-client.ts

```diff
@@ -46,6 +46,22 @@ interface StreamCallbacks {
   resetAnswer: () => void;
 }
 
+class TimeoutStateManager {
+  private timeouts: Set<ReturnType<typeof setTimeout>> = new Set();
+
+  public add(timeout: ReturnType<typeof setTimeout>) {
+    this.timeouts.add(timeout);
+  }
+
+  public remove(timeout: ReturnType<typeof setTimeout>) {
+    this.timeouts.delete(timeout);
+  }
```

Comment:
L135
L146
L153
We run `clearTimeout` in these places but we don't remove them from this state manager. Would it be safer to execute `timeouStateManager.remove()` and have remove defined like this:
```suggestion
  public remove(timeout: ReturnType<typeof setTimeout>) {
    clearTimeout(timeout);
    this.timeouts.delete(timeout);
  }
```

Reply:
yes, It make sense!